DEVELOPER_GUIDE.md

```diff
@@ -3,74 +3,185 @@
+## Getting started guide
 
-- [Accessibility developer guide (EUI Docs)](https://elastic.github.io/eui/#/guidelines/accessibility)
-- [SASS developer guide (EUI Docs)](https://elastic.github.io/eui/#/guidelines/sass)
+This guide is for any developer who wants a running local development environment where you can make, see, and test changes. It's opinionated to get you running as quickly and easily as possible, but it's not the only way to set up a development environment.
 
-## Getting Started
+If you're only interested in installing and running this project, please see the [Installing OpenSearch Dashboards](https://opensearch.org/docs/latest/install-and-configure/install-dashboards) instead.
 
-If you would like to install and run this project, please see the [Downloads Page](https://opensearch.org/downloads.html).
+If you're planning to contribute code (features or fixes) to this repository, great! Make sure to also read the [contributing guide](CONTRIBUTING.md).
 
-#### Prerequisites
+### Key technologies
 
-We recommend using [Node Version Manager](https://github.com/nvm-sh/nvm) to install
-the node version we need.
+OpenSearch Dashboards is primarily a node.js web application built in React. To effectively contribute you should be familiar with HTML ([usage guide](#html)), SASS styling ([usage guide](#sass-files)), TypeScript and JavaScript ([usage guide](#typescriptjavascript)), and React ([usage guide](#react)).
 
-### Bootstrap OpenSearch Dashboards
+### Prerequisites
+
+To develop on OpenSearch Dashboards, you'll need:
+
+- A [GitHub account](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account)
+- [`git`](https://git-scm.com/) for version control
+- [`node`](https://nodejs.org/), [`npm`](https://www.npmjs.com/), and [`yarn`](https://yarnpkg.com/) for building and running the project
+- A code editor of your choice, configured for JavaScript/TypeScript. If you don't have a favorite editor, we suggest [Visual Studio Code](https://code.visualstudio.com/)
+
+If you already have these installed or have your own preferences for installing them, skip ahead to the to the [Fork and clone OpenSearch Dashboards](#fork-and-clone-opensearch-dashboards) section.
+
+#### Install `git`
+
+If you don't already have it installed (check with `git --version`) we recommend following the [the `git` installation guide for your OS](https://git-scm.com/downloads).
+
+#### Install `node`
+
+We recommend using [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm) to install and manage different node versions, which may differ between release branches.
+
+1. Install nvm (as specified by the [`nvm` README](https://github.com/nvm-sh/nvm#installing-and-updating)): `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash`
```

Comment:
How so? If they follow the default `git` installation in the previous step, they will have GitBash installed, in which case this command works fine (as it did when I dry-ran all this on my windows laptop). Is it a matter of just clarifying that it assumes GitBash or WSL?

Reply:
GH is putting my comments in the wrong place; the comment was about installing `nvm`.

Reply:
We can maybe follow-up on this - when I went through the windows default git install in the previous step, it left me with an open GitBash terminal, and so I used the curl command above as-is to install `nvm`, and everything worked fine. But I definitely think we can refine the directions from the perspective of a more typical windows dev (because if I have to develop in windows, I'll personally always use WSL 😂 )